src/database/index.ts

```diff
@@ -0,0 +1,26 @@
+import { Database, IBaseRecord } from './db.interfaces';
+
+class InMemoryDatabase<T extends IBaseRecord> implements Database<T> {
+    private db: Record<string, T> = {};
+
+    async create(newValue: T): Promise<T | undefined> {
+        const itExists = Boolean(await this.getById(newValue.id));
```

Comment:
Consider storing the result of the `getById` and all similar operations into separate variable before applying any logic to improve readability.